### ISSUE TITLE
Adding a needless .into should produce a better error.

### ISSUE THREAD

**Issue by EvanCarroll** [rust-lang/rust#89204]
Suggested from [this question](https://stackoverflow.com/q/69303389/124486) on SO.

This code,

```rust
seq += rhs.into();
```

Will produce this error,

```rust
error[E0283]: type annotations needed
  --> src/sequence.rs:50:5
   |
19 | seq += rhs.into();
   |     ^^ ---------- this method call resolves to `T`
   |     |
   |     cannot infer type for type parameter `T`
   |
   = note: cannot satisfy `_: Into<u64>`
   = note: required because of the requirements on the impl of `AddAssign<_>` for `Sequence`
```

This can lead you down a rabbit hole. The right answer is far more simple: **remove the `.into()`**

```
seq += rhs;
```
I would like error to say (copied from SO),

> You have a double Into indirection. Since your type already implements AddAssign<T> where T: Into<u64>, then there is no need to add .into() to your right-hand member.

That's ideal. Alternatively telling me it could just work. (I'm still kind of confused why it doesn't), but the solution is far easier and nicer than either,

    seq += u64::from(rhs);
    Into<u64>::into(rhs);
